lib/improver/cli/time_lagged_ensembles.py

```diff
@@ -31,85 +31,42 @@
 # POSSIBILITY OF SUCH DAMAGE.
 """Script to run time-lagged ensembles."""
 
-import warnings
+from improver import cli
 
-import iris
 
-from improver.argparser import ArgParser
-from improver.utilities.load import load_cube
-from improver.utilities.save import save_netcdf
-from improver.utilities.time_lagging import GenerateTimeLaggedEnsemble
+@cli.clizefy
+@cli.with_output
+def process(*cubelist: cli.inputcube):
+    """Module to time-lag ensembles.
 
-
-def main(argv=None):
-    """Load in the arguments and ensure they are set correctly. Then run
-    the time-lagged ensembles on the input cubes.
-    """
-    parser = ArgParser(
-        description='This combines the realizations from different forecast '
-                    'cycles into one cube. It does this by taking an input '
-                    'CubeList containing forecasts from different cycles and '
-                    'merges them into a single cube, removing any metadata '
-                    'that does not match.')
-    parser.add_argument('input_filenames', metavar='INPUT_FILENAMES',
-                        nargs="+", type=str,
-                        help='Paths to input NetCDF files for the time-lagged '
-                        'ensemble to combine the realizations.')
-    parser.add_argument('output_file', metavar='OUTPUT_FILE',
-                        help='The output file for the processed NetCDF.')
-    args = parser.parse_args(args=argv)
-
-    # Load the cubes
-    cubes = iris.cube.CubeList([])
-    for filename in args.input_filenames:
-        new_cube = load_cube(filename)
-        cubes.append(new_cube)
-
-    # Process Cube
-    result = process(cubes)
-
-    # Save Cube
-    save_netcdf(result, args.output_file)
-
-
-def process(cubes):
-    """Module to run time-lagged ensembles.
-
-    This combines the realization from different forecast cycles into one cube.
-    It does this by taking an input Cubelist containing forecasts from
-    different cycles and merges them into a single cube, removing any
-    metadata that does not match.
+    Combines the realization from different forecast cycles into one cube.
+    Takes an input CubeList containing forecasts from different cycles and
+    merges them into a single cube.
 
     Args:
-        cubes (iris.cube.CubeList):
-            CubeList for the time-lagged ensemble to combine the realizations.
+        cubelist (iris.cube.CubeList):
```

Comment:
This will actually be a `list of iris.cube.Cube` - so we'll need to do `iris.cube.CubeList(cubelist)` inside the function somewhere.

Reply:
(To be honest, to save confusion, I'd probably rename the input to `cubes`, then have `cubelist = CubeList(cubes)` as required)

Reply:
It all works and there's nowhere in the code it needs to be a `CubeList`!!!  A list of cubes works fine for all of the called functions - as evinced by the fact that all the tests pass.  Should I just change all the docstrings to reflect this?  (This would potentially be quite a lot of docstrings given `load_cubelist` etc...)